photon-ml/src/main/scala/com/linkedin/photon/ml/algorithm/CoordinateDescent.scala

```diff
@@ -138,9 +156,9 @@ class CoordinateDescent(
               .setName(s"Updated model with coordinateId $coordinateId at iteration $iteration")
               .persistRDD(StorageLevel.INFREQUENT_REUSE_RDD_STORAGE_LEVEL)
               .materialize()
-
           case _ =>
         }
+        // We need to split out the following 2 match expressions: [[FactoredRandomEffectModel]] matches both
         oldModel match {
           case broadcastLike: BroadcastLike => broadcastLike.unpersistBroadcast()
           case _ =>
```

Comment:
XianXing - could you give more details about the rationale for using BroadcastLike sometimes: what does it try to achieve? Is it always the fixed effect model (and only that model) that can be BroadcastLike? Can it be not-BroadcastLinke? When/why would that happen (if it happens)?